File: Tuples/tuple.rs
use std::time::Instant;

fn main() {
    // Creación de la tupla
    let tup = (123, 456.789, "Hello, World!");

    // Medición de tiempo de acceso
    let start_time = Instant::now();
    for _ in 0..2000000 {
        let _ = tup.0;
        let _ = tup.1;
        let _ = tup.2;
    }
    let elapsed_time = start_time.elapsed().as_millis();
    println!("            _______  ");
    println!("          _/       \\_");
    println!("         / |       | \\");
    println!("        /  |__   __|  \\");
    println!("       |__/((o| |o))\\__|");
    println!("       |      | |      |");
    println!("       |\\     |_|     /|");
    println!("       | \\           / |");
    println!("        \\| /  ___  \\ |/");
    println!("         \\ | / _ \\ | /");
    println!("          \\_________/");
    println!("           _|_____|_");
    println!("      ____|_________|____");
    println!("     /                   \\"); 
    println!("╔═════════════════════════════╗");
    println!("║     Tuplas en Rust 🦀       ║");
    println!("║-----------------------------║");
    println!("║ Tiempo de compilación: {:.2}ms  ║", elapsed_time);
    println!("╚═════════════════════════════╝");
}

File: Tuples/tuple.py
import time

# Creación de la tupla
tup = (123, 456.789, "Hello, World!")

# Medición de tiempo de acceso
start_time = time.time()
for _ in range(2000000):
    _ = tup[0]
    _ = tup[1]
    _ = tup[2]

elapsed_time = (time.time() - start_time) * 1000  # Convertir a milisegundos
print("         /^\\/^\\")
print("       _|__|  O|")
print("\\/    /~     \\_/ \\")
print(" \\____|__________/  \\")
print("        \\_______      \\")
print("                `\\     \\                 \\")
print("                  |     |                  \\")
print("                 /      /                    \\")
print("                /     /                       \\\\")
print("              /      /                         \\ \\")
print("             /     /                            \\  \\")
print("           /     /             _----_            \\   \\")
print("          /     /           _-~      ~-_         |   |")
print("         (      (        _-~    _--_    ~-_     _/   |")
print(f"          \\      ~-____-~    _-~    ~-_    ~-_-~    /")
print(f"            ~-_           _-~          ~-_       _-~")
print(f"               ~--______-~                ~-___-~")
print("             ╔═════════════════════════════╗")
print("             ║    Tuplas en Python 󰌠       ║")
print("             ║-----------------------------║")
print(f"             ║ Tiempo de acceso: {elapsed_time:.2f} ms ║")
print("             ╚═════════════════════════════╝")
